devicetypes/sky-nie/child-button.src/child-button.groovy

```diff
@@ -0,0 +1,43 @@
+/**
+ *  Child Button v1.0 (CHILD DEVICE)
+ *
+ *  Author: 
+ *   winnie (sky-nie)
+ *
+ *  Changelog:
+ *
+ *    1.0 (03/16/2020)
+ *      - Initial Release
+ *
+ * Reference：
+ *  https://github.com/krlaframboise/SmartThings/blob/master/devicetypes/krlaframboise/component-button.src/component-button.groovy
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ */
+  
+metadata {
+	definition (name: "Child Button", namespace: "sky-nie", author: "winnie", ocfDeviceType: "x.com.st.d.remotecontroller") {
```

Comment:
If that child component is an integral part of the device - it Should be a Child Switch (with implemented functions:  `on(), off(), refresh()` and added `capability "Refresh"`)

```
void on() {
	parent.childOn()
}

void off() {
	parent.childOff()
}

def refresh() {
	parent.refreshChild()
} 

```
Addition functions in a parent DTH might be needed too.

Reply:
Oh, the button is only created for Paddles and is not always available (only for parts of devices), so we can’t use it as a child switch.